modules/@apostrophecms/modal/ui/apos/components/AposModal.vue

```diff
@@ -268,7 +272,18 @@ function trapFocus() {
 
   store.updateModalData(props.modalData.id, { elementsToFocus });
 
-  focusElement(props.modalData.focusedElement, props.modalData.elementsToFocus[0]);
+  const firstElementToFocus = findPriorityElementOrFirst(elementsToFocus);
+  const foundPriorityElement = firstElementToFocus.hasAttribute('data-apos-focus-priority');
+
+  focusElement(props.modalData.focusedElement, findPriorityElementOrFirst(elementsToFocus));
+
+  // Components render at various times and can't be counted on to be available on modal's mount
+  // Update the trap focus list until a data-apos-focus-priority element is found or the retry limit is reached
+  if (!foundPriorityElement && findPriorityFocusElementRetryMax.value > currentPriorityFocusElementRetry.value) {
+    await new Promise(resolve => setTimeout(resolve, 200));
```

Comment:
with 5 retries, it'll wait 1s for elements to be loaded.
is it enough?

Reply:
In my testing I haven't seen the retry count go above 2. Do you have a change to suggest? higher retry limit?